test/ziggurat/producer_test.clj

```diff
@@ -1,16 +1,14 @@
 (ns ziggurat.producer-test
   (:require [clojure.string :refer [blank?]]
-            [clojure.test :refer :all]
+            [clojure.test :refer [deftest is testing use-fixtures]]
             [clojure.test.check.generators :as gen]
             [ziggurat.config :refer [ziggurat-config]]
             [ziggurat.fixtures :as fix :refer [*producer-properties* *consumer-properties*]]
-            [ziggurat.producer :refer [producer-properties-map send kafka-producers
-                                       property->fn -send producer-properties]]
-            [ziggurat.streams :refer [start-streams stop-streams]]
+            [ziggurat.producer :refer [producer-properties-map send kafka-producers -send]]
             [ziggurat.tracer :refer [tracer]])
-  (:import (org.apache.kafka.clients.producer KafkaProducer ProducerRecord ProducerConfig)
-           [org.apache.kafka.streams.integration.utils IntegrationTestUtils]
-           [io.opentracing.contrib.kafka TracingKafkaProducer]))
+  (:import [io.opentracing.contrib.kafka TracingKafkaProducer]
+           [org.apache.kafka.clients.producer KafkaProducer]
+           [org.apache.kafka.streams.integration.utils IntegrationTestUtils]))
 
 (use-fixtures :once fix/mount-producer-with-config-and-tracer)
 
```

Comment:
Same as before:
Each namespace should have one test for backward compatibility - where we can check if older configs for that namespace are set properly.

And, one test for non-kafka-config-keys

Reply:
same as https://github.com/gojek/ziggurat/pull/219#discussion_r614615370